src/typechecker/FStar.TypeChecker.TcTerm.fsti

```diff
@@ -48,6 +48,7 @@ val tc_tot_or_gtot_term: env -> term -> term * lcomp * guard_t
 val tc_check_tot_or_gtot_term: env -> term -> typ -> string -> term * lcomp * guard_t
 val tc_tactic : typ -> typ -> env -> term -> term * lcomp * guard_t
 val tc_trivial_guard: env -> term -> term * lcomp
+val tc_local_attributes: env -> list term -> list term
```

Comment:
We discussed making it uniform to check all attributes the same way, can this then be renamed to `tc_attributes`, and we can call this for all attributes?

Reply:
I'm looking into this, and there are two issues:
 1. the existing codebase uses attributes on sigelts of various types, not only `unit`;
 2. type checking attributes also means dealing with universes, thus one cannot have universe-polymorphic attributes as an attribute (this breaks Steel [here](https://github.com/FStarLang/FStar/blob/df0c1bee44543026db3558757bb195fd5cbcd6da/ulib/experimental/Steel.ST.Util.fsti#L153) for instance).

Point (1) is easily dealt with: instead of type checking and expecting a `unit`, we can allow attributes of any type.
Point (2) seems more complicated to address to me; a solution would be to have a quoted term as an attribute, but it is a bit unfriendly.
I actually wonder if attributes should be type-checked at all.